core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/IoUtil.java

```diff
@@ -231,6 +242,62 @@ public static File downloadFile(File parent, String prefix, String suffix, URL u
         return file;
     }
 
+
+
+    /**
+     * Download file with redirection if got status 301, 302, 303.
+     * Will useful in case redirection http -> https
+     *
+     * @param parent
+     *         parent directory, may be <code>null</code> then use 'java.io.tmpdir'
+     * @param prefix
+     *         prefix of temporary file name, may not be <code>null</code> and must be at least three characters long
+     * @param suffix
+     *         suffix of temporary file name, may be <code>null</code>
+     * @param url
+     *         URL for download
+     * @return downloaded file
+     * @throws java.io.IOException
+     *         if any i/o error occurs
+     */
+    public static File downloadFileWithRedirect(File parent, String prefix, String suffix, URL url) throws IOException {
```

Comment:
Have you tried to allow redirects in the connection instead of maling this custom logic?